win32ss/user/ntuser/kbdlayout.c

```diff
@@ -650,14 +650,21 @@ IntImmActivateLayout(
         co_IntSendMessage(hImeWnd, WM_IME_SYSTEM, IMS_ACTIVATELAYOUT, (LPARAM)pKL->hkl);
         UserDerefObjectCo(pImeWnd);
     }
-    else if (pti->spDefaultImc)
+    else
     {
-        /* IME Activation is needed */
-        pti->pClientInfo->CI_flags |= CI_IMMACTIVATE;
+        /* Remember old keyboard layout to switch back for Chinese IMEs */
```

Comment:
I suppose it's not only for Chinese IMEs, but for any IME that would behave similarly.